py/desispec/scripts/zproc.py

```diff
@@ -350,53 +415,76 @@ def main(args=None, comm=None):
     timer.start('groupspec')
 
     nblocks, block_size, block_rank, block_num = \
-        distribute_ranks_to_blocks(len(all_spectros), rank=rank, size=size, log=log)
+        distribute_ranks_to_blocks(len(all_subgroups), rank=rank, size=size, log=log)
 
     if rank == 0:
-        splog = findfile('spectra', night=thrunight, tile=tileid,
-                         groupname=groupname, spectrograph=0,
-                         expid=expids[0],  # needed for perexp; ignored otherwise
-                         logfile=True)
-        os.makedirs(os.path.dirname(splog), exist_ok=True)
+        if groupname == 'healpix':
+            for hpix in args.healpix:
+                findfileopts['healpix'] = hpix
+                splog = findfile('spectra', spectrograph=0, logfile=True, **findfileopts)
+                os.makedirs(os.path.dirname(splog), exist_ok=True)
+        else:
+            splog = findfile('spectra', spectrograph=0, logfile=True, **findfileopts)
+            os.makedirs(os.path.dirname(splog), exist_ok=True)
         
     if comm is not None:
         comm.barrier()
 
     if block_rank == 0:
-        for i in range(block_num, len(all_spectros), nblocks):
+        for i in range(block_num, len(all_subgroups), nblocks):
             result, success = 0, True
-            spectro = all_spectros[i]
-            spectrafile = findfile('spectra', night=thrunight, tile=tileid,
-                                   expid=expids[0],
-                                   groupname=groupname, spectrograph=spectro)
-            splog = findfile('spectra', night=thrunight, tile=tileid,
-                             expid=expids[0],
-                             groupname=groupname, spectrograph=spectro,
-                             logfile=True)
-            coaddfile = findfile('coadd', night=thrunight, tile=tileid,
-                                 expid=expids[0],
-                                 groupname=groupname, spectrograph=spectro)
-            # generate list of cframes from dict of exposures, nights, and cameras
-            cframes = []
-            for (expid, night), cameras in expnight_dict.items():
-                for camera in cameras:
-                    if int(spectro) == int(camera[1]):
-                        cframes.append(findfile('cframe', night=night,
-                                                expid=expid, camera=camera))
-
             if groupname == 'healpix':
-                raise ValueError('groupname=healpix not yet supported; needs different desi_group_spectra options')
+                healpix = all_subgroups[i]
+                findfileopts['healpix'] = healpix
+
+                cframes = []
+                ii = hpixexp['HEALPIX'] == healpix
+                for night, expid, spectro in hpixexp['NIGHT', 'EXPID', 'SPECTRO'][ii]:
```

Comment:
Yeah, it's using a file system check instead of having an authoritative understanding of what is supposed to exist.  That isn't ideal, but I would like to save it for a future PR (or a clever postdoc contribution!) unless the exposure table updates for "if a camera is bad for every exposure of a tile on a night, flag the entire petal because stdstar will fail anyway" don't resolve all of the remaining corner case problems.

Reply:
Understood, I can have a look at it. The code for tile-based can be adapted, but depending on the timing of this PR it may not happen before this is merged. In any case what is there is fine. We just need to have other sanity checks to verify the cframes files exist, which we will have.